cellprofiler_core/utilities/image.py

```diff
@@ -301,16 +304,34 @@ def download_to_temp_file(url):
             Params={'Bucket': bucket_name, 'Key': key.replace("+", " ")}
         )
 
-    from urllib.request import urlopen
-    logging.info(f"Downloading image from {url}")
-    src = urlopen(url)
-    dest_file = tempfile.NamedTemporaryFile(suffix=ext, delete=False, dir=CP_TEMP_DIR.name)
-    try:
-        shutil.copyfileobj(src, dest_file)
-    except Exception as e:
-        logging.error(f"Unable to download image to temp file. {e}")
-        return None
-    finally:
-        dest_file.close()
-    return dest_file.name
+    elif scheme == 'gs':
```

Comment:
The updates to the GCS specific code look good!

I did notice that S3 urls won't actually get downloaded anymore due to the nesting. The old code is basically:
* if its an S3 URL
   * get the presigned version of that url so that we can download it just like any other url
* download the URL (whether it was an S3 URL or some other URL).

To fix it here, consider reordering the cases:
* if its a GCS URL
  * use the GCS client to download it
* else
  * if its an S3 URL
     * get the presigned version of that url so that we can download it just like any other url
  * download the URL (whether it was an S3 URL or some other URL).

Reply:
After this is fixed, its ready for others to review!

Reply:
Fixed!

Reply:
great! @bethac07 could you and your team please review these changes? Thank you.